src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java

```diff
@@ -39,32 +50,107 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
+            List<ExcludeDate> excludedDates = mutableEmptyListIfNull(hearingOptions.getExcludeDates());
+            excludedDates.add(ExcludeDate.builder().value(dateRange).build());
+            hearingOptions.setExcludeDates(excludedDates);
+        }
+    }
+
+    private void sendToJudge(String userAuthorisation, SscsCaseData sscsCaseData) {
+        if (sscsCaseData.getAppealNotePad() == null) {
+            sscsCaseData.setAppealNotePad(NotePad.builder().notesCollection(new ArrayList<>()).build());
+        }
+        sscsCaseData.getAppealNotePad().getNotesCollection()
+                .add(createPostponementRequestNote(userAuthorisation,
+                        sscsCaseData.getPostponementRequest().getPostponementRequestDetails()));
+        sscsCaseData.setInterlocReviewState(InterlocReviewState.REVIEW_BY_JUDGE.getId());
+        sscsCaseData.setInterlocReferralReason(InterlocReferralReason.REVIEW_POSTPONEMENT_REQUEST.getId());
+        sscsCaseData.setPostponementRequest(PostponementRequest.builder().unprocessedPostponementRequest(YesNo.YES)
+                .build());
+    }
+
+    private void addDirectionNotice(SscsCaseData caseData) {
+        DocumentLink url = caseData.getPreviewDocument();
+        SscsDocumentTranslationStatus documentTranslationStatus = caseData.isLanguagePreferenceWelsh() ? SscsDocumentTranslationStatus.TRANSLATION_REQUIRED : null;
+        footerService.createFooterAndAddDocToCase(url, caseData, AUDIO_VIDEO_EVIDENCE_DIRECTION_NOTICE,
+                Optional.ofNullable(caseData.getDateAdded()).orElse(LocalDate.now())
+                        .format(DateTimeFormatter.ofPattern("dd-MM-yyyy")),
+                caseData.getDateAdded(), null, documentTranslationStatus);
+    }
+
     private Note createPostponementRequestNote(String userAuthorisation, String details) {
         return Note.builder().value(NoteDetails.builder().noteDetail(details)
                 .author(userDetailsService.buildLoggedInUserName(userAuthorisation))
                 .noteDate(LocalDate.now().toString()).build()).build();
     }
+
+    private boolean isNotListable(PostponementRequest postponementRequest) {
+        return postponementRequest.getListingOption().equals(NOT_LISTABLE.getValue());
+    }
+
+    private boolean isReadyToList(PostponementRequest postponementRequest) {
+        return postponementRequest.getListingOption().equals(READY_TO_LIST.getValue());
+    }
+
+    private boolean isGrantPostponement(PostponementRequest postponementRequest) {
+        return postponementRequest.getActionPostponementRequestSelected().equals(GRANT.getValue());
+    }
+
+    private boolean isSendToJudge(PostponementRequest postponementRequest) {
+        return postponementRequest.getActionPostponementRequestSelected().equals(SEND_TO_JUDGE.getValue());
+    }
+
+    private static LocalDate getLocalDate(String dateStr) {
+        return LocalDate.parse(dateStr, DateTimeFormatter.ofPattern("dd-MM-yyyy"));
```

Comment:
Is this correct date formatter? I get a parsing error for this format. I think ccd has yyyy-MM-dd format. Also better to move this to a constant as this is used in three different places in this class.

Reply:
well spotted the wrong doc type. I had to create a new one as Postponement Request is the doc type for the doc in Action FE

Reply:
this is what I could find wide spread in tribunals

Reply:
Process A/V was my template

Reply:
I think dd-MM-yyyy format is used to write date values, but when you read date string coming from ccd, you might need to use yyyy-MM-dd. Please have a look at RequestHearingRecordingAboutToSubmitHandler, which is reading same date fields that you are reading here.

Reply:
fixed